src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -247,6 +247,7 @@ public DashboardWebApplication(Action<WebApplicationBuilder>? configureBuilder =
 
         _app.UseAuthorization();
 
+        _app.UseMiddleware<BrowserSecurityHeadersMiddleware>();
         _app.UseAntiforgery();
 
         _app.MapRazorComponents<App>().AddInteractiveServerRenderMode();
```

Comment:
You might want to check if you are targeting .NET 9.0. As compression is enabled by default and we setup an additional CSP policy value (ancestors-src: 'self'). 

I don't think it'll matter, as both policies will be applied and the most strict always win. Just letting you know in case you want to use a single header (you'll have to set one option in AddInteractiveServerRenderMode to null, and add it to your CSP policy header).

As per the other headers they look ok to me. It might be worth tightening up the CSP policy a bit more by avoiding unsafe-inline and using a nonce instead. Even though its only on CSS which is arguably way less problematic that on scripts.

Might also be worth hashing all the scripts and using hashes as opposed to `self`. That will protect you better against stored XSS if there is an encoding error.